src/PregReplace.php

```diff
@@ -137,8 +137,13 @@ public function getReplacement()
      */
     public function filter($value)
     {
-        if (! is_scalar($value) && ! is_array($value)) {
-            return $value;
+        if (! is_array($value)) {
```

Comment:
What happens if the array contains non-scalars?

Reply:
I will update this according to the solution decided below.